File: java/akka-scala/src/main/scala/com.asiafrank.learn.akka/example/Child.scala
package com.asiafrank.learn.akka.example

import akka.actor.Actor
import akka.event.Logging

/**
  * Child
  * <p>
  * </p>
  * Created at 2017/1/19 0019.
  *
  * @author zhangxf
  */
class Child extends Actor {
  val log = Logging(context.system, this)

  override def receive: Receive = {
    case "good" =>
      log.info("received good")
      sender() ! "end"
    case _      => log.info("child received unknown message")
  }
}

File: java/akka-java/src/main/java/com/asiafrank/akka/receivetimeout/Child.java
package com.asiafrank.akka.receivetimeout;

import akka.actor.Actor;
import akka.actor.Props;
import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;
import akka.japi.Creator;

/**
 * Child
 * <p>
 * Created by zhangxf on 12/7/2016.
 */
class Child extends UntypedActor {
    private final LoggingAdapter log = Logging.getLogger(getContext().system(), this);

    public static Props props() {
        return Props.create(new Creator<Actor>() {
            private static final long serialVersionUID = 1L;

            @Override
            public Actor create() throws Exception {
                return new Child();
            }
        });
    }

    @Override
    public void onReceive(Object message) throws Throwable {
        log.info("start");
        if (message.equals("start")) {
            Thread.sleep(3000);
            getSender().tell("end", getSelf());
        }
        log.info("end");
    }
}
